modules/vaos/app/controllers/vaos/v2/appointments_controller.rb

```diff
@@ -549,15 +552,36 @@ def fetch_drive_times(provider)
       def check_referral_usage(referral_id)
         check = appointments_service.referral_appointment_already_exists?(referral_id)
 
-        if check[:error]
-          { success: false, json: { message: "Error checking appointments: #{check[:failures]}" },
-            status: :bad_gateway }
-        elsif check[:exists]
-          { success: false, json: { message: 'No new appointment created: referral is already used' },
-            status: :unprocessable_entity }
-        else
-          { success: true }
-        end
+        return referral_check_error_response(check[:failures]) if check[:error]
+        return referral_already_used_response if check[:exists]
+
+        { success: true }
+      end
+
+      def referral_check_error_response(failures)
+        {
+          success: false,
+          json: {
+            errors: [{
+              title: 'Appointment creation failed',
+              detail: "Error checking existing appointments: #{failures}"
+            }]
+          },
+          status: :bad_gateway
+        }
+      end
+
+      def referral_already_used_response
+        {
+          success: false,
+          json: {
+            errors: [{
+              title: 'Appointment creation failed',
+              detail: 'No new appointment created: referral is already used'
+            }]
+          },
+          status: :unprocessable_entity
+        }
```

Comment:
these would be better handled as raised `EpsServiceException`s that would be caught in the rescue block of the `create_draft` action but leaving as is for now because the amount of change required to do this will be